lib/global-styles.php

```diff
@@ -196,45 +200,74 @@ function gutenberg_experimental_global_styles_settings( $settings ) {
 	unset( $settings['gradients'] );
 
 	$resolver = new WP_Theme_JSON_Resolver();
-	$all      = $resolver->get_origin( $theme_support_data );
-	$base     = $resolver->get_origin( $theme_support_data, 'theme' );
+	$origin   = 'theme';
+	if (
+		gutenberg_experimental_global_styles_has_theme_json_support() &&
+		gutenberg_is_fse_theme()
+	) {
+		// Only lookup for the user data if we need it.
+		$origin = 'user';
+	}
+	$tree = $resolver->get_origin( $theme_support_data, $origin );
 
 	// STEP 1: ADD FEATURES
-	// These need to be added to settings always.
-	$settings['__experimentalFeatures'] = $all->get_settings();
+	//
+	// These need to be always added to the editor settings,
+	// even for themes that don't support theme.json.
+	// An example of this is that the presets are configured
+	// from the theme support data.
+	$settings['__experimentalFeatures'] = $tree->get_settings();
```

Comment:
This will still look up for the theme.json, although the theme doesn't have one. I'd like to squeeze this performance bit as well, but it was getting convoluted so I'd like to do it in a follow-up (it requires improving the `WP_Theme_JSON_Resolver` API for consumers). So, this PR improves a bit the situation but doesn't fix all the things.